Refactor firebase lookup endpoint

BenjaminSsempala opened the pull request:
**_WHAT DOES THIS PR DO?_**
Refactors lookup endpoint to;
- return exists as a boolean value.
- return "user does not exist" message when user does not exist and Phone number or email address is invalid.

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

- Jira cards
    - [MOB-146](https://airqoteam.atlassian.net/browse/MOB-146)
- GitHub issues
    - Closes #<enter GitHub issue number here>

**_HOW DO I TEST OUT THIS PR?_**
- Checkout this branch and navigate to src/auth-service
- Run npm install and npm run dev-mac.
- Test out using this [firebase lookup endpoint.](https://airqo-engineering.postman.co/workspace/internal~7a8e9fad-2d13-4d27-af27-f1f4e9a114bf/request/12513372-b0dbf57c-1598-4783-9fef-72e08d7252de)
**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
[Firebase Lookup](https://airqo-engineering.postman.co/workspace/internal~7a8e9fad-2d13-4d27-af27-f1f4e9a114bf/request/12513372-b0dbf57c-1598-4783-9fef-72e08d7252de)
**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**

**_SCREENSHOTS_**
<img src ="https://user-images.githubusercontent.com/86492979/208880744-6ff32844-05c6-4f9c-b11e-64bfef9a5746.png" height =600 width =600>

**_ARE THERE ANY RELATED PRs?_**

Baalmart commented on src/auth-service/controllers/join.js:
```diff
@@ -91,7 +91,7 @@ const join = {
         return badRequest(
           res,
-          "bad request errors",
+          "User does not exist",
```
Thanks @BenjaminSsempala for this.

I am not sure about this change.....
Is this "function" for many other use cases that are not just about users?

**Context:**
Auth service handles more than just users, it does handle:
1. Inquiries
2. candidates
3. organisations/networks and
4. defaults

Please help shed more light on this?

BenjaminSsempala replied:
Hello @Baalmart 
I made this change inside the `lookupfirebase` function which I assumed is only utilized to look up users in firebase for the mobile app, although I could be wrong. With that assumption in mind, I thought to make the message more specific since it would only be for users. Also, some additional context may be acquired from the information in this [jira card](https://airqoteam.atlassian.net/browse/MOB-146).

Baalmart replied:
Thanks @BenjaminSsempala for the clarification, much appreciated!

Baalmart reviewed (commented):
Thanks @BenjaminSsempala , I have just left some  small comment.

Also, please shed more light about the changes? Could you use the PR description or review comments to share more context here?

Thanks!

BenjaminSsempala commented on src/auth-service/controllers/join.js:
```diff
@@ -125,6 +125,7 @@ const join = {
             success: true,
             message: result.message,
             user: result.data,
+            exists: true,
```
Return a boolean value for exists for when a user already exists.

BenjaminSsempala commented on src/auth-service/controllers/join.js:
```diff
@@ -137,7 +138,8 @@ const join = {
             success: false,
-            message: result.message,
+            message: "User does not exist",
+            exists: false,
```
Return the message as "user does not exist" and the boolean value exists as false

BenjaminSsempala commented:
> Thanks @BenjaminSsempala , I have just left some small comment.
> 
> Also, please shed more light about the changes? Could you use the PR description or review comments to share more context here?
> 
> Thanks!

Thank you @Baalmart , I've updated te description and also added some review comments.

Baalmart reviewed (approved):
These changes look good to me, thanks @BenjaminSsempala !
